client/utils.go

```diff
@@ -2,15 +2,22 @@ package client
 
 // Paginate returns the correct starting and ending index for a paginated query,
 // given that client provides a desired page and limit of objects and the handler
-// provides the total number of objects. If the start page is invalid, non-positive
-// values are returned signaling the request is invalid.
-//
-// NOTE: The start page is assumed to be 1-indexed.
+// provides the total number of objects. The start page is assumed to be 1-indexed.
+// If the start page is invalid, non-positive values are returned signaling the
+// request is invalid; it returns non-positive values if limit is non-positive and
+// defLimit is negative.
 func Paginate(numObjs, page, limit, defLimit int) (start, end int) {
-	if page == 0 {
+	if page <= 0 {
 		// invalid start page
 		return -1, -1
-	} else if limit == 0 {
+	}
+
+	// fallback to default limit if supplied limit is invalid
+	if limit <= 0 {
+		if defLimit < 0 {
```

Comment:
hmm shouldn't this be `defLimit <= 0`?

Reply:
We'll, in case defLimit is 0 and applied as the actual limit though I think 0 results would be returned. That should be ok.